### DISCUSSION TITLE
simulation error

### DISCUSSION THREAD

**Discussion by MohammadHadiAlizadeh** [OpenModelica/OpenModelica#11253] [Q&A] [answered]
Hello,

I'm currently running a simulation and encountering simulation-time errors such as "Fatal error on line ..., column ..: Opening and ending tag mismatch" after approximately t=170 hours. I have also attached a log file where the solver appears to be failing around the same time. However, OpenModelica (OM) displays results up to the designated final time of 1e6 seconds, even though it does not indicate that "the simulation finished successfully" (LOG_Success is enabled). I'm using OM version 1.21, the official release.
[SCD_nosurgev6_rebL1.log](https://github.com/OpenModelica/OpenModelica/files/12715990/SCD_nosurgev6_rebL1.log)

**Comment by casella** [comment]
@AnHeuermann can you explain why this happens?

**Comment by MohammadHadiAlizadeh** [comment]
Attaching more information.
![tag_mismatch](https://github.com/OpenModelica/OpenModelica/assets/88594092/cfdc3e48-4d30-4169-a644-177e464cd404)
![tag_mismatch_finaltime](https://github.com/OpenModelica/OpenModelica/assets/88594092/0a4ba5e1-5653-4720-9240-498033743918)

**Comment by lucas-mior** [comment]
I have had a similar error when multiple debug output options are active. Notice the message `Reached display limit. ...`.
Try deactivating some debug options and try again.

**Comment by casella** [comment]
@adeas31 this also happened to me several times.

I searched the entire OMC source code for the "tag mismatch" string but I could not find it. I suspect this error comes from some library routine that OMEdit is calling to process the simulation executable. Is that the case, and if so do you have any idea why this happens? Thanks!

**Reply by perost** [reply]
It's a common XML error message, but I'm not sure where it's coming from.

**Reply by adrpo** [reply]
As far as I know, the log messages that the simulation executable sends to OMEdit is XML and if there is something wrong in there, OMEdit would not be able to parse the output,  @adeas31 should know more.

**Reply by adeas31** [reply]
OMEdit uses an incremental parser. This looks like XML parsing error due to not well formed XML. I can debug more if you can provide me a test model. Maybe the simulation runtime returns an unexpected end tag or maybe the xml text is not escaped properly.

**Reply by MohammadHadiAlizadeh** [reply]
@adeas31 Just sent you an email containing a sample model.

**Comment by casella** [comment]
See ticket #12386, the "tag mismatch" error message is definitely a bug that needs to be resolved. Once that is fixed, we can hopefully figure out why the simulation actually fails.

**Comment by adeas31** [comment, marked as answer]
#12391 fixes the issue. No more error messages are shown and the simulation runs fine.
